### DISCUSSION TITLE
Pull a specific project rather than looping?

### DISCUSSION THREAD

**Discussion by roadSurfer** [pmckeown/dependency-track-maven-plugin#342] [Ideas]
I've noticed that the plugin pulls all the projects from the DepTrack server and then filters them by name & version.
Would it not be more efficient to simply ask DepTrack for the specific project?

I have a [forked branch of the code](https://github.com/roadSurfer/dependency-track-maven-plugin/blob/project-lookup/src/main/java/io/github/pmckeown/dependencytrack/project/ProjectAction.java) for which I could issue a pull request if you like.

**Comment by pmckeown** [comment]
The project was initially built for use in CI/CD pipelines in a microservices architecture where we wanted seamless integration to Dep Track during the pipeline, even for new projects that don't exist in Dep Track yet.  So the plugin was set up to use the maven coordinates as the primary mechanism for identifying projects in Dependency Track, as that is the primary identifier in the maven build lifecycle.  This meant we didn't need to encode an ID from another system into the Maven POM file to be able to use the plugin.  

So we buy some complexity in implementation for simplicity in use.

**Reply by roadSurfer** [reply]
Ah - I think I may have been unclear; I didn't mean using the project UUID, just sticking with name/version.

It's really just a question of who does the work and getting DepTrack to do it means the database indexes can be leant on, along with lower network traffic. The signature of `ProjectAction.getProject` doesn't change at all.

Unless, of course, I have misunderstood you.
